core/src/main/java/io/grpc/CallCredentials.java

```diff
@@ -65,6 +72,7 @@
    * @param applier The outlet of the produced headers. It can be called either before or after this
    *        method returns.
    */
+  @ExperimentalApi("https//github.com/grpc/grpc-java/issues/1914")
   void applyRequestMetadata(
```

Comment:
Talked offline. It seems we're both okay with adding a useless method that implementors have to implement to make things clear. Adding a `iAmUsingUnstableApi()` method.